IPython/terminal/embed.py

```diff
@@ -83,6 +96,8 @@ def __init__(self, **kw):
 
         super(InteractiveShellEmbed,self).__init__(**kw)
 
+        frame = sys._getframe(InteractiveShellEmbed._stack_frames)
+        self._call_location_id = '%s:%s' % (frame.f_code.co_filename, frame.f_lineno)
```

Comment:
I can try to take a hash of the stack-trace, like `'-'.join('%s:%s' % (frame.f_code.co_filename, frame.f_lineno) for frame in sys.get_frames()`, which will be more unique depending on how you access the line where embed is called. Might be more robust. But I don't see how to deactivate a specific location permanently, cleanly another way.